src/OpenSSL/SSL.py

```diff
@@ -699,8 +714,98 @@ def set_default_verify_paths(self):
 
         :return: None
         """
+        # This function will attempt to load certs from both a cafile and
+        # capath that are set at compile time. However, it will first check
+        # environment variables and, if present, load those paths instead
         set_result = _lib.SSL_CTX_set_default_verify_paths(self._context)
         _openssl_assert(set_result == 1)
+        # After attempting to set default_verify_paths we need to know whether
+        # to go down the fallback path.
+        # First we'll check to see if any env vars have been set. If so,
+        # we won't try to do anything else because the user has set the path
+        # themselves.
+        dir_env_var = _ffi.string(
+            _lib.X509_get_default_cert_dir_env()
+        ).decode("ascii")
+        file_env_var = _ffi.string(
+            _lib.X509_get_default_cert_file_env()
+        ).decode("ascii")
+        if not self._verify_env_vars_set(dir_env_var, file_env_var):
+            # If no env vars are set next we want to see if any certs were
+            # loaded. For a cafile this is simple and we can just ask how many
+            # objects are present. However, the cert directory (capath) is
+            # lazily loaded and num will always be zero so we need to check if
+            # the dir exists and has valid file names in it to cover that case.
+            num = self._check_num_store_objects()
```

Comment:
Yeah, I didn't think of that until just now so I'm not sure I've thought through the implications, but it seems like rather than trying to detect if a system has successfully loaded roots maybe we're better off only modifying our behavior when we know we're running under a manylinux1 cryptography build.